executor/src/witgen/machines/fixed_lookup_machine.rs

```diff
@@ -231,17 +230,18 @@ impl<'a, T: FieldElement> FixedLookup<'a, T> {
         &mut self,
         mutable_state: &MutableState<'a, T, Q>,
         identity_id: u64,
-        rows: &RowPair<'_, 'a, T>,
         outer_query: OuterQuery<'a, '_, T>,
     ) -> EvalResult<'a, T> {
         let right = self.connections[&identity_id].right;
 
-        if outer_query.left.len() == 1 && !outer_query.left.first().unwrap().is_constant() {
+        if outer_query.parameters.len() == 1
+            && !outer_query.parameters.first().unwrap().is_constant()
+        {
             if let Some(column_reference) = try_to_simple_poly(&right.expressions[0]) {
                 // Lookup of the form "c $ [ X ] in [ B ]". Might be a conditional range check.
                 return self.process_range_check(
-                    rows,
-                    outer_query.left.first().unwrap(),
+                    outer_query.range_constraints,
+                    outer_query.parameters.first().unwrap(),
                     AlgebraicVariable::Column(column_reference),
                 );
             }
```

Comment:
Should we respect the range constraints in the rest of the code or is it just a hint? Please document in the trait function.

Reply:
Not sure what you mean. Document in `Machine::process_plookup` (when are we going to rename this haha)?

I think it would be nice to respect them and could even lead to a unique result when it's not unique just from the known values. But seems like it's not needed in practice. Seems a out of the scope of this PR to fix this.

Reply:
ah true, because we get a single result anyway. ok!